src/libs/actions/TaxRate.ts

```diff
@@ -119,10 +182,299 @@ function clearTaxRateError(policyID: string, taxID: string, pendingAction?: Pend
     Onyx.merge(`${ONYXKEYS.COLLECTION.POLICY}${policyID}`, {
         taxRates: {
             taxes: {
-                [taxID]: {pendingAction: null, errors: null},
+                [taxID]: {pendingAction: null, errors: null, errorFields: null},
             },
         },
     });
 }
 
-export {createPolicyTax, clearTaxRateError, getNextTaxCode, getTaxValueWithPercentage};
+type TaxRateEnabledMap = Record<string, Pick<TaxRate, 'isDisabled' | 'errors' | 'pendingAction' | 'pendingFields' | 'errorFields'>>;
+
+function setPolicyTaxesEnabled(policyID: string, taxesIDsToUpdate: string[], isEnabled: boolean) {
+    const policy = allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${policyID}`];
+    const originalTaxes = {...policy?.taxRates?.taxes};
+
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: taxesIDsToUpdate.reduce<TaxRateEnabledMap>((acc, taxID) => {
+                            acc[taxID] = {
+                                isDisabled: !isEnabled,
+                                pendingFields: {isDisabled: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE},
+                                errorFields: {isDisabled: null},
+                            };
+                            return acc;
+                        }, {}),
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    taxRates: {
+                        taxes: taxesIDsToUpdate.reduce<TaxRateEnabledMap>((acc, taxID) => {
+                            acc[taxID] = {isDisabled: !isEnabled, pendingFields: {isDisabled: null}, errorFields: {isDisabled: null}};
```

Comment:
We don't need to update the `isDisabled` value since it was already done in `optimisticData`
```suggestion
                            acc[taxID] = {pendingFields: {isDisabled: null}, errorFields: {isDisabled: null}};
```